nemo/collections/nlp/modules/common/transformer/transformer_decoders.py

```diff
@@ -168,7 +172,12 @@ def forward(
         cached_mems_list = [memory_states]
 
         for i, layer in enumerate(self.layers):
-            decoder_states = layer(decoder_states, decoder_attn_mask, memory_states, encoder_states, encoder_attn_mask)
+            if i == len(self.layers) - 1:
```

Comment:
I think doing it this way will add an extra element to `cached_mems_list` so it will have len(self.layers) + 1 elements in the list. I think the way to do it should be to iterate until len(self.layers) - 1. Then check if i == len(self.layers) - 2 and then do the last transformer block + LN and then add that to cached mems list. My guess is this is why the export test is failing as well?

Reply:
hmm yes looks like it gonna have extra hidden states mem from LN. Are you suggesting that we should make **last transformer block + LN** as one component/layer?

Reply:
I think you can still keep the layers separate like you've done right now, but just apply last the transformer block + LN before adding to the `cached_mems_list`.

Reply:
Got it. On it.